src/container/srv_target.c

```diff
@@ -323,21 +276,40 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *msecs)
 	if (epoch_max >= cont->sc_aggregation_max)
 		epoch_max = cont->sc_aggregation_max - 1;
 
+	if (param->ap_max_eph_get) {
+		uint64_t max_eph = param->ap_max_eph_get(cont);
+
+		epoch_max = min(epoch_max, max_eph);
+		if (epoch_min > epoch_max) {
+			/**
+			 * For VOS aggregation, max might be 0, if EC
+			 * aggregation does not broadcast boundary yet.
```

Comment:
sorry, seems I lost the point, VOS agg MUST happen after EC agg and within the EC agg's AGG EPOCH boundary, right? in this patch you want to make VOS agg and EC agg be independent?

Reply:
yes, VOS aggregation  must respect the EC aggregation boundary. This patch is only for separate them into different ULT for clearing the logic, and fixing some bugs, like VOS aggregation will not be blocked if EC aggregation is stuck, EC aggregation will not be triggered if snapshot is being deleted, etc.